File: backend/src/mypm/presentation/api/schemas/task.py
"""Task API 스키마."""

from __future__ import annotations

import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from mypm.domain.tasks.entities import TaskStatus


class TaskCreateSchema(BaseModel):
    title: str = Field(..., description="할 일 제목")
    description: str | None = Field(None, description="할 일 설명")
    due_date: datetime.date | None = Field(None, description="마감일")


class TaskUpdateSchema(BaseModel):
    title: str | None = Field(None, description="할 일 제목")
    description: str | None = Field(None, description="할 일 설명")
    status: TaskStatus | None = Field(None, description="할 일 상태")
    due_date: datetime.date | None = Field(None, description="마감일")


class TaskResponseSchema(BaseModel):
    id: UUID
    title: str
    description: str | None
    status: TaskStatus
    due_date: datetime.date | None
    retrospective_id: UUID | None

    class Config:
        from_attributes = True

File: frontend/src/types/task.ts
export type TaskStatus = 'todo' | 'in_progress' | 'done' | 'blocked'

export interface Task {
  id: string
  title: string
  description: string | null
  status: TaskStatus
  due_date: string | null
  retrospective_id: string | null
}

export interface TaskCreatePayload {
  title: string
  description?: string | null
  due_date?: string | null
}

export interface TaskUpdatePayload {
  title?: string
  description?: string | null
  status?: TaskStatus
  due_date?: string | null
}

export const TASK_STATUS_LABELS: Record<TaskStatus, string> = {
  todo: '할 일',
  in_progress: '진행중',
  done: '완료',
  blocked: '차단됨',
}

export const TASK_STATUS_ORDER: TaskStatus[] = ['todo', 'in_progress', 'done', 'blocked']


